test/test_base.py

```diff
@@ -176,3 +194,26 @@ def get_scenario_return_code(self, scenario_name):
         if len(scns) == 0:
             raise Exception(f"Scenario {scenario_name} not found in running scenarios")
         return scns[0]["return_code"]
+
+
+def assert_equal(thing1, thing2, *args):
+    if thing1 != thing2 or any(thing1 != arg for arg in args):
+        raise AssertionError(
+            "not({})".format(" == ".join(str(arg) for arg in (thing1, thing2) + args))
+        )
+
+
+def assert_log(log_message, expected_msgs, unexpected_msgs=None) -> bool:
+    if unexpected_msgs is None:
+        unexpected_msgs = []
+    assert_equal(type(expected_msgs), list)
+    assert_equal(type(unexpected_msgs), list)
+
+    found = True
+    for unexpected_msg in unexpected_msgs:
+        if re.search(re.escape(unexpected_msg), log_message, flags=re.MULTILINE):
+            raise AssertionError(f"Unexpected message found in log: {unexpected_msg}")
```

Comment:
I was able to sneak an unexpected message into the log without test failure:

```diff
diff --git a/test/dag_connection_test.py b/test/dag_connection_test.py
index 6e87b79..b5df653 100755
--- a/test/dag_connection_test.py
+++ b/test/dag_connection_test.py
@@ -33,7 +33,7 @@ class DAGConnectionTest(TestBase):
         self.log_expected_msgs = [
             "Successfully ran the connect_dag.py scenario using a temporary file"
         ]
-        self.log_unexpected_msgs = ["Test failed."]
+        self.log_unexpected_msgs = ["Successfully ran the connect_dag.py scenario using a temporary file"]
         self.wait_for_all_scenarios()
         self.assert_log_msgs_passed()
 
```

Reply:
Thanks. Pushed a fix commit.